File: core/src/com/t2wonderland/kurona/Objects/Candy.java
package com.t2wonderland.kurona.Objects;

import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;
import com.t2wonderland.kurona.Interfaces.IGameObject;

public class Candy implements IGameObject {

    float WIDTH = 1f;
    float HEIGHT = 1f;

    Vector2 _size = new Vector2(WIDTH, HEIGHT);
    Vector2 _position;
    Rectangle _bounds;

    @Override
    public void setInitialPosition(float x, float y) {
        this._position = new Vector2(x, y);
        this._bounds = new Rectangle(x - WIDTH / 2, y - HEIGHT / 2, WIDTH, HEIGHT);
    }

    @Override
    public Vector2 getSize() {
        return _size;
    }

    @Override
    public Vector2 getPosition() {
        return _position;
    }

    @Override
    public Rectangle getBounds() {
        return _bounds;
    }
}

File: core/src/com/t2wonderland/kurona/Objects/Candy.kt
package com.t2wonderland.kurona.Objects

import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.math.Vector2
import com.t2wonderland.kurona.Interfaces.IStaticObject

class Candy : IStaticObject {

    private var WIDTH = 1f
    private var HEIGHT = 1f

    var _position: Vector2 = Vector2()
    var _bounds: Rectangle = Rectangle()
    var _size : Vector2 = Vector2(WIDTH, HEIGHT)

    override fun setInitialPosition(x: Float, y: Float) {
        this._position.x = x
        this._position.y = y
        this._bounds = Rectangle(x - WIDTH / 2, y - HEIGHT / 2, WIDTH, HEIGHT)
    }

    override fun getSize(): Vector2 {
        return _size
    }

    override fun setPosition(position: Vector2) {
        _position = position
    }

    override fun getPosition(): Vector2 {
        return _position
    }

    override fun getBounds(): Rectangle {
        return _bounds
    }
}
